cmd/operator-sdk/bundle/output.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bundle
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+)
+
+const (
+	infoLevel  = "INFO"
+	errorLevel = "ERROR"
+)
+
+// Log is an object which is used to return results in the JSON format
+type Log struct {
+	Type    string `outputFormat:"type"`
+	Message string `outputFormat:"message"`
+	Error   string `outputFormat:"error,omitempty"`
+}
+
+// OutPut represents the final result
+type OutPut struct {
+	Passed bool  `outputFormat:"passed"`
+	Logs   []Log `outputFormat:"logs"`
+}
+
+// newLog return a new *Log based on the info provided
+func newLog(msg string, err error) Log {
+	log := new(Log)
+	log.Type = infoLevel
+	log.Message = msg
+	if err != nil {
+		log.Error = err.Error()
+		log.Type = errorLevel
+	}
+	return *log
+}
+
+// printJSON will print the output in JSON format
+func (o *OutPut) printJSON() {
+	o.Passed = true
+	for _, i := range o.Logs {
+		if i.Type == errorLevel {
```

Comment:
A little bit concerned about the `==` comparison. The reason is that errorLevel is defined as a string rather than an integer or something that can express severity level in numbers. I was hoping that if errorLevel is set to "INFO", the filter will allow all warnings, errors and info alike logs. If errorLevel is set to "ERROR", that means no warnings or info level logs will be retained. Makes sense?

Reply:
removed